### ISSUE TITLE
On click `is-active-link` class always being added to wrong href tag

### ISSUE THREAD

**Issue by dbjpanda** [tscanlin/tocbot#145]
Here is an example url https://dbjpanda.me/dbms/fundamentals/introduction .
Click on any title inside table of content and you will see the `is-active-link` will be added to another href tag that is below the real active link.  
```
.is-active-link{
color: green;
}
```

![Alt Text](https://media.giphy.com/media/YQN9tkPhqWfyBo4YBb/giphy.gif)

[Here](https://github.com/dbjpanda/kb-elementor/blob/master/assets/js/toc-headings.js#L40) is the direct use of default configuration of tocbot.js so I don't think any issue with script. Heading offset is set to 1.

**Comment by tscanlin** [comment]
I think it isn't scrolling properly because of images that are loaded at the top of the page. Could you try deferring the init of tocbot by a couple seconds?

**Comment by dbjpanda** [comment]
If you see the source, you can find in the bottom if the page, tocbot script is already deferred there. 

![Screen Shot 2019-08-03 at 1 34 07 AM](https://user-images.githubusercontent.com/17483092/62395859-1e9bf980-b58f-11e9-8bdf-a206bc58f136.png)

Is it what you are saying ? To load the script after everything loaded?

**Comment by dbjpanda** [comment]
I found, there is no mistake in my side but in the tocbot script. 
Please check your demo site : https://tscanlin.github.io/tocbot 
Click on `Steps to publish` and you can see `is-active-link` class is being added to `License` instead of `Steps to Publish`. 
It seems while there are no children of an active `li`, the class is being added to underneath `li`. 
@tscanlin if possible please let me your opinion asap. 30+ users (https://wordpress.org/plugins/kb-elementor/ ) are waiting for the issue to be solved. :)

**Comment by tscanlin** [comment]
I verified this issue and have a branch with a fix open.
try out: #147 
you should be able to test it locally too

**Comment by dbjpanda** [comment]
@tscanlin Thanks. PR #147 works. Just two quick questions.

Shouldn't it be hard coded? If we set it to false, then it is not going to work as expected. Then what is use case of this option.
```
// For headings inside relative or absolute positioned containers within content.
  hasInnerContainers: true,
```
Which one do you call as "containers"?

**Comment by tscanlin** [comment]
I don't think all sites need it, if you don't have relatively positioned containers it shouldn't be needed. Also changing this default would be a breaking change and I'd like to avoid that. But I did update the example in the readme and I'm sorry it was an issue for you. The container was changed to relatively positioned in that PR (see link) but the option wasn't added to the init of tocbot in the docs, so thanks for bringing this to my attention so I could fix it. Let me know if I can clarify anything else!

https://github.com/tscanlin/tocbot/issues/135
https://github.com/tscanlin/tocbot/pull/136/files#diff-abbb3c1ecb85ad3731d54bb866a5cfcfR18
